app/uk/gov/hmrc/customs/declaration/services/BatchFileUploadUpscanNotificationBusinessService.scala

```diff
@@ -0,0 +1,90 @@
+import uk.gov.hmrc.customs.declaration.logging.DeclarationsLogger
+import uk.gov.hmrc.customs.declaration.model._
+import uk.gov.hmrc.customs.declaration.model.actionbuilders.HasConversationId
+import uk.gov.hmrc.customs.declaration.repo.BatchFileUploadMetadataRepo
+
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+@Singleton
+class BatchFileUploadUpscanNotificationBusinessService @Inject()(repo: BatchFileUploadMetadataRepo,
+                                                                 connector: FileTransmissionConnector,
+                                                                 config: DeclarationsConfigService,
+                                                                 logger: DeclarationsLogger) {
+
+  def persistAndCallFileTransmission(ready: UploadedReadyCallbackBody)(implicit r: HasConversationId): Future[Unit] = {
+    repo.update(
+      ready.reference,
+      CallbackFields(ready.uploadDetails.fileName, ready.uploadDetails.fileMimeType, ready.uploadDetails.checksum)
+    ).flatMap{
+      case None =>
+        val errorMsg = s"database error - can't find record with file reference ${ready.reference}"
+        logger.error(errorMsg)
+        Future.failed(new IllegalStateException(errorMsg))
+      case Some(metadata) =>
+        maybeFileTransmission(ready.reference, metadata) match {
+          case None =>
+            val errorMsg = s"database error - can't find file with file reference ${ready.reference}"
+            logger.error(errorMsg)
+            Future.failed(new IllegalStateException(errorMsg))
+          case Some(fileTransmission) =>
+            connector.send(fileTransmission).map { _ =>
+              logger.info(s"successfully called file transmission service $fileTransmission")
+              ()
+            }
+      }
+    }
+  }
+
+  private def maybeFileTransmission(fileReference: FileReference, md: BatchFileUploadMetadata): Option[FileTransmission] = {
+    for {
+      (bf, ftf) <- maybeFileTransmissionFile(fileReference, md)
+    } yield FileTransmission(
+      FileTransmissionBatch(md.batchId, md.fileCount),
+      new URL(config.batchFileUploadConfig.fileTransmissionBaseUrl),
+      ftf,
+      FileTransmissionInterface("DEC64", "1.0.0"),
+      Seq("DeclarationId" -> md.declarationId.toString, "Eori" -> md.eori.toString, "ContentType" -> bf.documentType.toString).map(t => FileTransmissionProperty(name = t._1, value = t._2))
+    )
+  }
+
+  private def maybeFileTransmissionFile(fileReference: FileReference, md: BatchFileUploadMetadata): Option[(BatchFile, FileTransmissionFile)] = {
+    for {
+      bf <- md.files.find(bf => bf.reference == fileReference)
+      cb <- bf.maybeCallbackFields
+    } yield (
+      bf,
+      FileTransmissionFile(
+        bf.reference,
+        cb.name,
+        cb.mimeType,
+        cb.checksum,
+        bf.location,
+        bf.sequenceNumber
+      )
+    )
+  }
+
+}
```

Comment:
General comment on above - my personal preference is to not use this many abbreviations. bf, ftf, md, cb... Makes reading the code more difficult than it needs to be. But, if you prefer this...

Reply:
done